plugin/src/test/java/org/jenkinsci/plugins/structs/describable/DescribableModelTest.java

```diff
@@ -725,6 +725,29 @@ public ToBeRemoved() {
         }
     }
 
+    @Issue("JENKINS-43337")
+    @Test
+    public void ambiguousSimpleName() throws Exception {
+        AmbiguousContainer container = new AmbiguousContainer(new FirstAmbiguous.CommonName("first"),
+                new UnambiguousClassName("second"));
+
+        UninstantiatedDescribable ud = DescribableModel.uninstantiate2_(container);
+
+        Object o = ud.toMap().get("ambiguous");
+        assertTrue(o instanceof Map);
+        Map<String,Object> m = (Map<String,Object>)o;
+
+        // Make sure the ambiguous class is fully qualified.
+        assertEquals(FirstAmbiguous.CommonName.class.getName(), m.get("$class"));
```

Comment:
Could probably be tested a lot more compactly using the `roundTrip` test utility, but this works.